### ISSUE TITLE
[QST]Is there any difference between different test data sets of tpc?

### ISSUE THREAD

**Issue by Matrix-World** [NVIDIA/spark-rapids#1436]
**What is your question?**
@abellina 
Sorry, I have been confused about the tests in the integration_tests/src/main/scala/com/nvidia/spark/rapids/tests. There are three types of tests, such as tpch, tpcds and tpcbb. They represent different tests, can you explain the difference between these three different tests? For example, what is the difference in the test data structure, and what is special about the executed SQL statement under the three different tests. In other words, what are the respective concerns of the three tests?
Looking forward to your reply.
Wish you a Happy New Year~

**Comment by abellina** [comment]
@Matrix-World 

The tests you pointed at are based on the [TPCH](http://www.tpc.org/tpch/), [TPC-DS](http://www.tpc.org/tpcds/), and [TPCx-BB](http://www.tpc.org/tpcx-bb/) benchmarks. We set them up in our repository such that tests could be triggered automatically by our CI system. It is really just a way for us to exercise our code regularly, and a convenient way to trigger benchmarks.

We don't change the SQL part of the query in any case, but we will adapt the datatypes to get parts of our plugin to work on the GPU. The only type change I am aware of is Decimal to double and BigInt to long.

Some benchmarks like TPCx-BB have more than just SQL (some of it is SQL for preprocessing simply to run java code afterwards, which is normally included in a TPCx-BB run), but they are not in active development and the source isn't available to get it up to spec, as it was built for older versions of Spark. We decided to focus on the SQL components of the benchmarks, especially since that is what we are accelerating, and have added other benchmarks that are more actively used today (TPC-DS) for better comparisons.

Hope this helps!
